solidity/contracts/PhasedEscrow.sol

```diff
@@ -71,3 +71,82 @@ contract CurveRewardsEscrowBeneficiary is Ownable {
+/// @notice A beneficiary contract that can receive a withdrawal phase from a
+///         PhasedEscrow contract. Immediately stakes the received tokens on a
+///         designated BeaconBackportRewardsEscrowBeneficiary contract.
+contract BeaconBackportRewardsEscrowBeneficiary is Ownable {
+    IERC20 public token;
+    IStakerRewards public stakerRewards;
+
+    constructor(IERC20 _token, IStakerRewards _stakerRewards) public {
+        token = _token;
+        stakerRewards = _stakerRewards;
+    }
+
+    function __escrowSentTokens(uint256 amount) external {
+        token.approve(address(stakerRewards), amount);
+        stakerRewards.receiveApproval(address(this), amount, address(token), "");
+    }
+}
+
+/// @title BeaconRewardsEscrowBeneficiary
+/// @notice A beneficiary contract that can receive a withdrawal phase from a
+///         PhasedEscrow contract. Immediately stakes the received tokens on a
+///         designated BeaconRewardsEscrowBeneficiary contract.
+contract BeaconRewardsEscrowBeneficiary is Ownable {
+    IERC20 public token;
+    IStakerRewards public stakerRewards;
+
+    constructor(IERC20 _token, IStakerRewards _stakerRewards) public {
+        token = _token;
+        stakerRewards = _stakerRewards;
+    }
+
+    function __escrowSentTokens(uint256 amount) external {
+        token.approve(address(stakerRewards), amount);
+        stakerRewards.receiveApproval(address(this), amount, address(token), "");
+    }
+}
+
+/// @title ECDSABackportRewardsEscrowBeneficiary
+/// @notice A beneficiary contract that can receive a withdrawal phase from a
+///         PhasedEscrow contract. Immediately stakes the received tokens on a
+///         designated ECDSABackportRewardsEscrowBeneficiary contract.
+contract ECDSABackportRewardsEscrowBeneficiary is Ownable {
+    IERC20 public token;
+    IStakerRewards public stakerRewards;
+
+    constructor(IERC20 _token, IStakerRewards _stakerRewards) public {
+        token = _token;
+        stakerRewards = _stakerRewards;
+    }
+
+    function __escrowSentTokens(uint256 amount) external {
+        token.approve(address(stakerRewards), amount);
+        stakerRewards.receiveApproval(address(this), amount, address(token), "");
+    }
+}
+
+/// @title ECDSARewardsEscrowBeneficiary
+/// @notice A beneficiary contract that can receive a withdrawal phase from a
+///         PhasedEscrow contract. Immediately stakes the received tokens on a
+///         designated ECDSARewardsEscrowBeneficiary contract.
+contract ECDSARewardsEscrowBeneficiary is Ownable {
+    IERC20 public token;
+    IStakerRewards public stakerRewards;
+
+    constructor(IERC20 _token, IStakerRewards _stakerRewards) public {
+        token = _token;
+        stakerRewards = _stakerRewards;
+    }
+
+    function __escrowSentTokens(uint256 amount) external {
+        token.approve(address(stakerRewards), amount);
+        stakerRewards.receiveApproval(address(this), amount, address(token), "");
+    }
```

Comment:
I do agree having 4 separate contracts is helpful as it's more transparent, easier to track where tokens go, and less chances of making a stupid mistake during the deployment. On the other hand, they have exactly the same code. Maybe we can extract the code to `StakerRewardsBeneficiary` contract and have 4 contracts inherit from it? This way, we can also simplify tests as we'll be testing only `StakerRewardsBeneficiary`.

Reply:
Sure, I think it's a good idea. Done.